core/src/data/rasterSource.cpp

```diff
@@ -77,100 +91,127 @@ RasterSource::RasterSource(const std::string& _name, std::unique_ptr<DataSource>
                            TextureOptions _options, TileSource::ZoomOptions _zoomOptions)
     : TileSource(_name, std::move(_sources), _zoomOptions),
       m_texOptions(_options) {
-    m_textures = std::make_shared<Cache>();
 
+    m_textures = std::make_shared<Cache>();
     m_emptyTexture = std::make_shared<Texture>(m_texOptions);
+
+    GLubyte pixel[4] = { 0, 0, 0, 0 };
+    auto bpp = _options.bytesPerPixel();
+    m_emptyTexture->setPixelData(1, 1, bpp, pixel, bpp);
+
+    Feature rasterFeature;
+    rasterFeature.geometryType = GeometryType::polygons;
+    rasterFeature.polygons = { { {
+                {0.0f, 0.0f},
+                {1.0f, 0.0f},
+                {1.0f, 1.0f},
+                {0.0f, 1.0f},
+                {0.0f, 0.0f}
+            } } };
+    rasterFeature.props = Properties();
+
+    m_tileData = std::make_shared<TileData>();
+    m_tileData->layers.emplace_back("");
+    m_tileData->layers.back().features.push_back(rasterFeature);
```

Comment:
Needs only to be initialized when `generateGeometry`

Reply:
Ok this fixes the crash, but we do not need to set this `Feature` for all `RasterSources`. As your comment mentioned we only need this when a raster source is actually a primary tile source.

Reply:
Yeah I would say negligible overhead for now - RasterSource needs some serious work anyway :)

Reply:
Thanks for fixing this! :D